src/Result.ts

```diff
@@ -8,8 +8,8 @@ export interface ResultMatchPattern<T, U, E, V>
 
 export interface Result<T, E>
 {
-    is_ok(): boolean;
-    is_err(): boolean;
+    is_ok(): this is Ok<E>;
```

Comment:
Great idea. The generic type should be T for is_ok `is_ok(): this is Ok<T>;`. Nevermind, I will merge this pull request, update this and Option.ts as well.

Thank you for your contribution~

Reply:
Darn't! Sorry, it was copy-pasta. You're very welcome :)